sources/packages/backend/libs/test-utils/src/mocks/ssh-service-mock.ts

```diff
@@ -41,28 +41,43 @@ export function createSSHServiceMock(
 }
 
 /**
- * Get the parameters provided to the put method of the SSH client that represents the
- * remote file path and the data that would be uploaded to the SFTP in a real scenario.
+ * Get the parameters provided to the put method of the SSH client that
+ * represents the data that would be uploaded to the SFTP in a real scenario.
  * @param sshClientMock SSH mocked client.
  * @returns file name and file content of the supposed-to-be uploaded file.
  */
 export function getUploadedFile(
   sshClientMock: DeepMocked<Client>,
 ): UploadedFile {
-  const uploadedFile = {} as UploadedFile;
+  const [uploadedFile] = getUploadedFiles(sshClientMock);
+  return uploadedFile;
+}
+
+/**
+ * Get the parameters provided to the put method of the SSH client that represents the
+ * remote file(s) path and the data that would be uploaded to the SFTP in a real scenario.
+ * @param sshClientMock SSH mocked client.
+ * @returns file(s) name and file(s) content of the supposed-to-be uploaded file(s).
+ */
+export function getUploadedFiles(
+  sshClientMock: DeepMocked<Client>,
+): UploadedFile[] {
   if (!sshClientMock.put.mock.calls.length) {
     throw new Error(
       "SSH client mock was not invoked which means that there was no attempt to upload a file.",
     );
   }
-  const [[input, remoteFilePath]] = sshClientMock.put.mock.calls;
-  if (input) {
-    uploadedFile.fileLines = input.toString().split(END_OF_LINE);
-  }
-  if (remoteFilePath) {
-    uploadedFile.remoteFilePath = remoteFilePath.toString();
-  }
-  return uploadedFile;
+  return sshClientMock.put.mock.calls.map((call) => {
+    const uploadedFile = {} as UploadedFile;
+    const [input, remoteFilePath] = call;
```

Comment:
Not a show stopper, just a suggestion. I believe we can remove this const and pass it in the call itself.

return sshClientMock.put.mock.calls.map(([input, remoteFilePath]) => {

Reply:
Changed.